coffee/src/CoffeeMachineSimulation.java

```diff
@@ -0,0 +1,212 @@
+        milkPerCup[ESPRESSO] = 0;
+        moneyPerCup[ESPRESSO] = 4;
+
+        waterPerCup[LATTE] = 350;
+        coffeePerCup[LATTE] = 20;
+        milkPerCup[LATTE] = 75;
+        moneyPerCup[LATTE] = 7;
+
+        waterPerCup[CAPPUCCINO] = 200;
+        coffeePerCup[CAPPUCCINO] = 12;
+        milkPerCup[CAPPUCCINO] = 100;
+        moneyPerCup[CAPPUCCINO] = 6;
+
+        setReady();
+    }
+
+    /**
+     * Prints the state of the coffee-machine.
+     */
+    private void remaining()
+    {
+        System.out.println("The coffee machine has:");
+
+        System.out.println(water + " of water");
+        System.out.println(milk + " of milk");
+        System.out.println(coffee + " coffee beans");
+        System.out.println(cups + " of disposable cups");
+        System.out.println("$" + money + " of money");
+
+    }
+    /**
+     * Check for resources to make at list one cup of coffee.
+     *
+     * @param coffeeType Coffee type code (1 - espresso, 2 - latte, 3 - cappuccino).
+     *
+     * @return Has resources or not.
+     */
+    private boolean hasResources(int coffeeType)
+    {
+        boolean has = true;
+
+        if(water < waterPerCup[coffeeType]) {
+            System.out.println("Sorry, not enough water!");
+            has = false;
+        }
+
+        if(coffee < coffeePerCup[coffeeType]) {
+            System.out.println("Sorry, not enough coffee!");
+            has = false;
+        }
+
+        if(milk < milkPerCup[coffeeType]) {
+            System.out.println("Sorry, not enough milk!");
+            has = false;
+        }
+
+        if(cups == 0) {
+            System.out.println("Sorry, not enough cups!");
+            has = false;
+        }
+
+        return has;
+    }
+
+    private void buy(int index)
+    {
+        --index;
+
+        if (hasResources(index)) {
+            System.out.println("I have enough resources, making you a coffee!");
+
+            water -= waterPerCup[index];
+            milk -= milkPerCup[index];
+            coffee -= coffeePerCup[index];
+            money += moneyPerCup[index];
+
+            cups--;
+        }
+    }
+
+    private void take()
+    {
+        System.out.println("I gave you $" + money);
+        money = 0;
+    }
+
+    private void setReady()
+    {
+        state = State.Ready;
+        System.out.println("Write action (buy, fill, take, remaining, exit): ");
+        System.out.print("> ");
+    }
+
+    private void askForNumber(String prompt)
+    {
+        System.out.println(prompt);
+        System.out.print("> ");
+    }
+
+    /**
+     *check to see if the input is integer representation
+     */
+    private boolean isInteger(String input)
+    {
+        for (int i = 0; i < input.length(); ++i) {
+            if (!Character.isDigit(input.charAt(i))) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private void action(String input)
+    {
+        boolean isInt = isInteger(input);

```

Comment:
If isInt is mandatory just check it one time and print error massage, without checking it in all if cases

Reply:
Done